### PR TITLE
fix: remove dependency on react-router, use react-router-dom instead.

### PR DESCRIPTION
**Pull request by fengelniederhammer**
See https://github.com/remix-run/react-router/blob/f0b886bce7dab2e2f37f53fe58cd6c7ab0660e25/packages/react-router/README.md

The problem appeared in #729

### CODE REVIEW

**Review by chaoran-chen — APPROVED** [review]
Thanks for the fix!

File: src/__tests__/App.test.tsx
Comment (fengelniederhammer) [inline src/__tests__/App.test.tsx:8]: This test needs the api mocks from the mutation overview branch, so I rebased it on top of it. We'll have to merge this after the base branch is merged.

**Pull request merged**
